### DISCUSSION TITLE
The View does't update after the state is changed.

### DISCUSSION THREAD

**Discussion by hellotimw** [pointfreeco/swift-composable-architecture#1292] [Q&A]
This is my state code, where submit ButtonDisabled is used to calculate whether the button is disabled or not. This is mainly based on whether the tagName of the corresponding input box is not empty.

```
struct TagEditorState: Equatable {
  @BindableState var tagName = ""
  @BindableState var focusedField: Field?

  var submitButtonDisabled: Bool {
    self.tagName.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
  }
}
```

This part of the code is my submit button, where the disabled part does not become available based on my input. What is the problem and how should I change my code?

```
var submitButton: some View {
    WithViewStore(self.store) { viewStore in
      Button(action: {
        viewStore.send(.submitButtonTapped)
      }, label: {
        Text(viewStore.tag == nil ? "button.submit".localized() : "button.save".localized())
      })
      .buttonStyle(PrimaryButtonStyle())
      .disabled(viewStore.submitButtonDisabled)
      .sheet(isPresented: viewStore.binding(get: \.membershipIntroductionSheetIsPresented, send: TagEditorAction.setMembershipIntroductionSheet(isPresented: ))) {
        MembershipIntroductionSheet()
      }
    }
  }
```

**Comment by tgrapperon** [comment]
Nothing stands particularly out in what you're showing, excepted that there is no `.tag` in `TagEditorState`, so we might be missing problematic bits elsewhere. This view's `Button` should be disabled or enabled when `tagName` changes from empty to non-empty. So I would make sure that this value is indeed changing. The `PrimaryButtonStyle` is also maybe participating to give the impression that the `Button` is not disabled. If this is a custom `ButtonStyle`, you should probably expose an `@Environment(\.isEnabled) var isEnabled` in your modifier, and adapt styling accordingly. 
So I would test the issue by simply removing the styling to begin with, and check if this is only a styling issue.
Assuming this `ViewStore` is indeed operating on a complete `TagEditorState`, I would also initially put a `Text("\(viewStore.tagName.count)")` somewhere to check what it sees.
